pycodestyle.py

```diff
@@ -1544,11 +1557,17 @@ def ambiguous_identifier(logical_line, tokens):
             if text in idents_to_avoid:
                 ident = text
                 pos = start
-        # function parameter definitions
-        if is_func_def:
-            if text in idents_to_avoid:
-                ident = text
-                pos = start
+        # function / lambda parameter definitions
+        if is_func_def or is_lambda_def:
+            next_token = tokens[index + 1] if index < len(tokens) - 1 else None
+            next_text = next_token[1] if next_token else None
+            if (
+                next_text and next_text in ':,=)' and
+                prev_text in ('lambda', ',', '*', '**', '(')
+            ):
```

Comment:
I would write this as:

```python
if index < len(tokens) - 1 and tokens[index + 1][1] in ... and ...:
```

that way you take advantage of short circuiting and don't do unnecessary work

Reply:
oh yeah, that's simpler too. nice